Content.Server/Nutrition/EntitySystems/DrinkSystem.cs

```diff
@@ -162,7 +163,7 @@ public void UpdateAppearance(EntityUid uid, DrinkComponent component)
     /// </summary>
     private bool TryDrink(EntityUid user, EntityUid target, DrinkComponent drink, EntityUid item)
     {
-        if (!HasComp<BodyComponent>(target))
+        if (!HasComp<BodyComponent>(target) || HasComp<BorgChassisComponent>(target))
```

Comment:
Rather than specifically check for borgs, you should create a more general component that allows mobs to eat and/or drink and check for that. This should check should also be added to the food system when attempting to eat

Reply:
That would require adding the new component to every mob prototype that should be able to eat / drink, no? The current food and drink system currently just check if the mob has a body, then a stomach that can support consuming the item. Although the drink system currently assumes there is a stomach present which is where this bug arises from.

Reply:
> That would require adding the new component to every mob prototype that should be able to eat / drink, no? 

Yup. But if you add it to a base mob, it cut down the number of changes required

> The current food and drink system currently just check if the mob has a body, then a stomach that can support consuming the item. Although the drink system currently assumes there is a stomach present which is where this bug arises from.

If you want, you can fix the drink system so that the entity really does needs a stomach instead

Reply:
I think that would be a simpler fix, it brings the two systems more in line with one another. The `FoodSystem` does currently check for an `UnremovableComponent` though, which is what actually blocks borgs from eating something in their hand before it gets to the `StomachComponent` check.
The entity requiring a stomach is a longer term solution however.

Reply:
> Rather than specifically check for borgs, you should create a more general component that allows mobs to eat and/or drink and check for that

There was a refactor which was combining HungerSystem and ThirstSystem into a single SatiationSystem #27481
Sadly the PR was never finished, but this would have been a good solution for this imo

Reply:
Sorry for the lengthy delay on making the requested changes. I removed the explicit borg chassis check and added a stomach component check. This falls in line with how the FoodSystem prevents borgs from eating, which prevents any entity without a stomach from ingesting food.